[lennoxtr] iP

lennoxtr opened the pull request.

Commit:
Add Gradle support

lennoxtr committed:
initial commit

lennoxtr committed:
Add "Add and List" feature.

Add Pair.java and TaskList.java

lennoxtr committed:
Add "Mark and Unmark" feature

lennoxtr committed:
Implement ToDos, Events, Deadlines

Change tasks to class instead of Pair<Event, Boolean>

lennoxtr committed:
Implement Deleting tasks

lennoxtr committed:
Implement Exception Handling

lennoxtr committed:
Implement Save feature

lennoxtr committed:
Implement Java Codestyle

lennoxtr committed:
Implement more OOP, Package

lennoxtr committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-Level-7

lennoxtr committed:
Add Gradle Support, Add Codestyle

lennoxtr committed:
Add JUnit tests for TaskList

lennoxtr committed:
Update directory structure to deconflict

lennoxtr committed:
Implement Packaging

SPWwj commented on src/main/java/UserInteration.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.Scanner;
+class UserInteraction {
+    public void chat() {
+        Scanner sc = new Scanner(System.in);
+        Greeting greeting = new Greeting();
+        System.out.println(greeting.toString());
+        String isPlaying = sc.nextLine();
+        if (isPlaying.equals("NO")) {
+            greeting = new Greeting(0);
+            Event nextEvent = greeting.toNext();
+            System.out.println(nextEvent.toString());
+        } else if (isPlaying.equals("YES")) {
+            greeting = new Greeting(1);
+            Event nextEvent = greeting.toNext();
+            System.out.println(nextEvent.toString());
+            while (nextEvent.getStatus() == false) {
+                nextEvent = nextEvent.toNext();
+                System.out.println(nextEvent.toString());
+            }
+        } else {
+            while (!isPlaying.equals("YES") && !isPlaying.equals("NO")) {
+                System.out.println("_".repeat(22));
+                System.out.println("FOCUS, HUMAN. " +
+                        "YOU ARE TO ENTER INPUT WITH FULL CAPS.");
+                System.out.println("_".repeat(22));
+                isPlaying = sc.nextLine();
+            }
+            if (isPlaying.equals("NO")) {
+                greeting = new Greeting(0);
+                Event end = greeting.toNext();
+                System.out.println(end.toString());
+            } else {
+                greeting = new Greeting(1);
+                Event nextEvent = greeting.toNext();
+                System.out.println(nextEvent.toString());
+                while (nextEvent.getStatus() == false) {
+                    nextEvent = nextEvent.toNext();
+                    System.out.println(nextEvent.toString());
+                }
+            }
```
Perhaps can apply OOP here to make this function shorter and easier to read :grin:

SPWwj commented on src/main/java/Ending.java:
```diff
@@ -0,0 +1,17 @@
+class Ending extends Event {
```
Perhaps you may use nouns form for this class name

SPWwj replied:
Perhaps you may use nouns form for this class name :smiley:
```suggestion
Ending
```

SPWwj commented on src/main/java/Greeting.java:
```diff
@@ -0,0 +1,30 @@
+        this.status = -1;
+    }
+
+    public Greeting(int inPlay) {
```
Maybe can use a more sensible name for this :grin:
```suggestion
    inPlay
```

SPWwj commented on src/main/java/Greeting.java:
```diff
@@ -0,0 +1,30 @@
+        this.status = inPlay;
+    }
+
+    public Event toNext() {
```
Maybe can be more clearer on the nameing, like nextEvent or toNextEvent?
```suggestion
    public Event toNext() {
```

SPWwj commented on src/main/java/UserInputException.java:
```diff
@@ -0,0 +1,73 @@
+import java.util.List;
+class UserInputException extends Exception {
+    static void checkUserInput(String input, int currentSize) throws DukeException {
+        String[] inputSplitArray = input.split(" ");
```
Perhaps a more intuitive variable name order here like splitInputArray?
```suggestion
         inputSplitArray
```

SPWwj commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.ArrayList;
+import java.util.List;
+
+class TaskList {
```
Try to add more Javadoc comments

SPWwj commented on src/main/java/DoTask.java:
```diff
@@ -0,0 +1,137 @@
+import java.util.Scanner;
+import java.util.List;
+class DoTask extends Event {
+    boolean firstGreet;
```
What is firstGreet means?

SPWwj commented on src/main/java/DoTask.java:
```diff
@@ -0,0 +1,137 @@
+class DoTask extends Event {
+    boolean firstGreet;
+    String lastCommand;
+    TaskList taskList;
```
removedTask look like an action, why you choose to store it as a string datatype?

SPWwj commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        return this;
+    }
+
+    public Task unMark() {
```
Good job very sensible name!

SPWwj reviewed (commented):
LGTM, almost. Just a few nits to fix.
Improve on naming the variable and classes

lennoxtr committed:
Add Java Doc

lennoxtr committed:
Change date and time from String to LocalDateTime

lennoxtr committed:
Merge branch 'branch-Level-8' into branch-Level-7, resolve conflicts

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/io/input/exception/DukeException.java
#	src/main/java/duke/io/input/exception/UserInputException.java
#	src/main/java/duke/io/input/ui/UserInteraction.java
#	src/main/java/duke/util/Task.java
#	src/main/java/duke/util/TaskList.java
#	src/main/java/duke/util/service/Deadline.java
#	src/main/java/duke/util/service/ScheduledEvent.java
#	src/main/java/duke/util/service/ToDo.java
#	src/main/java/duke/workflow/DoTask.java
#	src/main/java/duke/workflow/Ending.java
#	src/main/java/duke/workflow/Event.java
#	src/main/java/duke/workflow/Greeting.java

cyiting commented on src/main/java/duke/io/input/ui/UserInteraction.java:
```diff
@@ -0,0 +1,101 @@
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.List;
+import duke.util.Storage;
```
Maybe you can group your import statements based on the packages?

cyiting commented on src/main/java/duke/workflow/DoTask.java:
```diff
@@ -0,0 +1,198 @@
+     * @return a new event that follows from the last user input
+     */
+
+    public Event toNext() {
```
Maybe you could include another class to parse the user inputs?

cyiting commented on src/main/java/duke/workflow/DoTask.java:
```diff
@@ -0,0 +1,198 @@
+                    if (words.get(0).equals("MARK")) {
+                        return new DoTask(false, nextTask, this.taskList.markDone(Integer.valueOf(words.get(1)) - 1));
+                    }
+                    if (words.get(0).equals("UNMARK")) {
```
Maybe you could use if-else statements instead of multiple ifs?

cyiting commented on src/main/java/duke/workflow/Greeting.java:
```diff
@@ -0,0 +1,61 @@
+     *          {@code DoTask} if the user wants to use Duke.
+     */
+
+    public Event toNext() {
```
`getNextEvent` may be a better method name than `toNext` according to the coding standards

cyiting commented on src/main/java/duke/workflow/Greeting.java:
```diff
@@ -0,0 +1,61 @@
+ * and ask whether he/ she would like to begin using Duke.
+ */
+
+public class Greeting extends Event {
```
I like how you have classes for your workflow.

cyiting reviewed (commented):
LGTM except for a few nits. Javadoc comments are clear too.

lennoxtr committed:
Implement Find, optimize TaskList

lennoxtr committed:
Fixed Codestyle

lennoxtr committed:
Update build.gradle for javafx

lennoxtr committed:
Implement GUI

lennoxtr committed:
Improve GUI Response

lennoxtr committed:
Add assert at mark/ unmark pipeline of Duke

Remove redundant code in Duke

lennoxtr committed:
Revert "Add assert at mark/ unmark pipeline of Duke"

This reverts commit 3a85840fb8629b38ecdc6bd81a5c72958ee5f2ef.

lennoxtr committed:
Add assert at mark/e umake pipeline in Duke

Remove javafx redudant code from Duke

lennoxtr committed:
Refactor code to improve code quality

lennoxtr committed:
Merge pull request #2 from lennoxtr/A-Assert

A assert

lennoxtr committed:
Fix implementation of loading file

Add customized Duke's greeting if user has existing progress

lennoxtr committed:
Fix implementation of loading file

Add customized Duke's greeting if user has used Duke before

lennoxtr committed:
Refactor code to debug new loading progress implementation

lennoxtr committed:
Fix merging issue

lennoxtr committed:
Merge pull request #3 from lennoxtr/A-Assert

Fix implementation of loading file

lennoxtr committed:
Revert "Fix merging issue"

This reverts commit bb600aab899243c4a960e1037e44b3234920b1ba.

lennoxtr committed:
Merge branch 'master' of https://github.com/lennoxtr/ip

lennoxtr committed:
no message

lennoxtr committed:
Refactor code to implement more OOP

Create Parser class to parse tasks

Change toNextEvent() function in Event classes to work with GUI

lennoxtr committed:
Extract out printing code to implement UserInterface

Refactor code so that GUI and CMI version can use the same UserInterface

Improve code readability

lennoxtr committed:
Implement 'find schedule on date' feature for Deadline class to find the tasks on a specified date

lennoxtr committed:
Include javadoc

Reconfigure launcher to run both CLI and GUI versions of Duke

lennoxtr committed:
implement saving feature for GUI Duke

lennoxtr committed:
Refactor code to improve code quality of loadProgress method in Storage

lennoxtr committed:
Refactor code in Storage to improve code quality

lennoxtr committed:
Change saveProgress method of Storage to create an output directory and save file to that directory

lennoxtr committed:
Update docs with program photos

lennoxtr committed:
Update docs with program photos

lennoxtr committed:
Rename photos to correct format

lennoxtr committed:
Update Readme file

lennoxtr committed:
Update Readme file

lennoxtr committed:
Merge branch 'master' into issue

# Conflicts:
#	docs/README.md

lennoxtr committed:
Revert "Update Readme file"

This reverts commit 8886fcfc48860ad7d3b8ae15f60626b17caf708f.

lennoxtr committed:
Update Readme.md

lennoxtr committed:
Update product screenshots

lennoxtr committed:
Implment printing error message to GUI version of Duke

lennoxtr committed:
Fixed window auto resizing after input

lennoxtr committed:
Add saving message for GUI Duke

lennoxtr committed:
Update product screenshots

lennoxtr committed:
Update product screenshot

lennoxtr committed:
Update product screenshot

lennoxtr committed:
Fix documentation problem of "Schedule" feature
